### ISSUE TITLE
`Function.prototype.toString` revision

### ISSUE THREAD

**Issue by mathiasbynens** [tc39/ecma262#664]
https://github.com/tc39/Function-prototype-toString-revision by @michaelficarra

Please tag this issue with the “web reality” label. Thanks!

**Comment by domenic** [comment]
Let's leave this open until the proposal has been merged into the main spec, so that people looking at https://github.com/tc39/ecma262/labels/web%20reality can get a better view.

**Comment by littledan** [comment]
I don't think @michaelficarra 's proposal reflects current web reality, as it is not what any particular browser currently ships, as far as I know.

**Comment by ljharb** [comment]
My understanding is that there is a significant percentage that intersects web reality, and for the rest, there isn't cross-browser agreement.

@michaelficarra could you post/point to the data from your presentation about that?

**Comment by michaelficarra** [comment]
@ljharb 

![screen shot 2016-08-18 at 03 22 20](https://cloud.githubusercontent.com/assets/218840/17770661/c13ba716-64f3-11e6-82d2-e2736e9e283a.png)

Yeah, I don't think that the Function.prototype.toString proposal differs from web reality. It just makes even stronger guarantees than both what was specified and what exists in implementations.

**Comment by michaelficarra** [comment]
By the way, this is why we have stage 3. We're waiting on implementation feedback to confirm that this is both technically feasible and web compatible before moving it to stage 4.

**Comment by bakkot** [comment]
fwiw, I believe the proposal [disagrees from web reality](https://github.com/tc39/Function-prototype-toString-revision/issues/17) in its treatment of source text occurring prior to the formal parameters in normal function expressions and declarations.

**Comment by michaelficarra** [comment]
@bakkot That's right. I should have said that I believe it is *web compatible*. So technically different, but not in a way that will break a meaningful number of web pages. But we'll have to wait for implementation feedback (hint, nudge, wink, etc.) before we know this to be true.

**Comment by leobalter** [comment]
@bakkot That change makes sense as it places the proposal consistent with it self.

As @ljharb mentioned: "... there is a significant percentage that intersects web reality, ...", which seems reasonable.

**Comment by bakkot** [comment]
Sure. Just wanted to clarify that "for the rest, there isn't cross-browser agreement" wasn't quite right.

On V8's end, including source text prior to the name would probably introduce some otherwise unnecessary overhead into every function, which we're not eager to do.
